tests/main/snap-session-agent-service-control/task.yaml

```diff
@@ -33,6 +33,9 @@ prepare: |
         snap alias test-snapd-curl.curl curl
     fi
 
+    echo "Ensure session agent service is ready to reply"
+    retry-tool curl --unix-socket "${USER_RUNTIME_DIR}/snapd-session-agent.socket" http://localhost/v1/service-control
```

Comment:
I think the real thing we're waiting on here is for the socket to exist: once that happens, connections should queue up activation of the underlying service.

If we assume that the user instance of systemd is in the process of starting up, what I think we want is to wait for `sockets.target` to start (or even just `snapd.session-agent.socket`).  I wonder if running the following (as the test user) would be enough:

    systemctl --user start sockets.target

I believe that should block until the given unit starts.